packages/inventories-exporter/src/main.js

```diff
@@ -0,0 +1,218 @@
+    this.logger.verbose('Starting Export')
+    if (this.exportConfig.format === 'csv') {
+      // open a stream to write csv from object
+      const csvOptions = {
+        headers: true,
+        delimiter: this.exportConfig.delimiter,
+      }
+      const csvStream = csv
+        .createWriteStream(csvOptions)
+        .transform((row: Inventory) => {
+          this.logger.verbose(`transforming row ${JSON.stringify(row)}`)
+          return InventoryExporter.inventoryMappings(row)
+        })
+      csvStream.pipe(outputStream)
+      this._fetchInventories(csvStream)
+        .then((): stream$Writable => csvStream.end())
+        .catch((e: Error) => {
+          outputStream.emit('error', e)
+        })
+    } else {
+      const jsonStream = JSONStream.stringify()
+      jsonStream.pipe(outputStream)
+      this._fetchInventories(jsonStream)
+        .then(() => {
+          if (outputStream !== process.stdout)
+            jsonStream.end()
+        })
+        .catch((e: Error) => {
+          outputStream.emit('error', e)
+        })
+    }
+  }
+
+  _fetchInventories (
+    outputStream: stream$Writable,
+  ): Promise<any> {
+    if (this.exportConfig.channelKey)
+      return this._resolveChannelKey(this.exportConfig.channelKey)
+        .then(channelId => this._makeRequest(outputStream, channelId))
+    return this._makeRequest(outputStream)
+  }
+
+  _makeRequest (outputStream: stream$Writable, channelId?: string) {
+    const query = this.reqBuilder
+      .inventory
+      .expand('custom.type')
+      .expand('supplyChannel')
+      .whereOperator('and')
+    if (this.exportConfig.queryString)
+      query.where(this.exportConfig.queryString)
+    if (channelId)
+      query.where(`supplyChannel(id="${channelId}")`)
+    const uri = query.build()
+    const request: ClientRequest = {
+      uri,
+      method: 'GET',
+    }
+    if (this.accessToken)
+      request.headers = {
+        Authorization: `Bearer ${this.accessToken}`,
+      }
+    return this.client.process(
+      request,
+      (payload: Object): Promise<any> => InventoryExporter._processFn(
+        payload.body.results,
+        outputStream,
+      ),
+      { accumulate: false },
+    )
+  }
+
+  _resolveChannelKey (channelKey: string): Promise<any> {
+    const queryString = `key="${channelKey}"`
+    const uri = this.reqBuilder
+      .channels
+      .where(queryString)
+      .build()
+    const request: ClientRequest = {
+      uri,
+      method: 'GET',
+    }
+    if (this.accessToken)
+      request.headers = {
+        Authorization: `Bearer ${this.accessToken}`,
+      }
+    return this.client.execute(request)
+      .then((result): Promise<any> => {
+        if (result.body && result.body.results.length)
+          return Promise.resolve(result.body.results[0].id)
+        return Promise.reject(
+          new Error('No data with channel key in CTP Platform'),
+        )
+      })
+  }
+
+  static _processFn (
```

Comment:
Are we to assume there would be no situation where we would need to return a rejected promise?